### PR TITLE
Add main queue default

### PR DESCRIPTION
**Pull request by KVTaniguchi**
The purpose of this PR is to add a default main queue method for Notifications, because 
os_dispatch

### CODE REVIEW

**Comment by ryang1428** [comment]
@KVTaniguchi I added the `public func post(notification: Notification) -> Void` function the other day for objective c compatibility. that defaults to the main queue. Id give that a shot first

**Comment by KVTaniguchi** [comment]
@ryang1428  

```public func post(notification: Notification) -> Void {
        post(notification)
  }```

the ```post(notification)``` in the objc compat method calls the swift version of ```post``` and doesn't invoke the ```URBNConvenience``` one, otherwise the the method call would have the param label in it and would like this:
```post(notification: notification)```

I've reopened this one with the correct method call.

File: Pod/Classes/NotificationConvenience.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ public extension NotificationCenter {
     
     // Needed for objc compatibility
     public func post(notification: Notification) -> Void {
-        post(notification)
+        post(notification:notification, queue: DispatchQueue.main)
```
Comment (KVTaniguchi) [inline Pod/Classes/NotificationConvenience.swift]: ```queue: DispatchQueue.main``` is required otherwise its calling itself.

Comment (ryang1428) [inline Pod/Classes/NotificationConvenience.swift]: ah yeah good catch.... I'm still in favor of going against the swifty way and going back to prefixing with `urbn_` 🙈

**Comment by KVTaniguchi** [comment]
@ryang1428  @rmigneco  any other changes you guys want to this?

**Comment by rmigneco** [comment]
assigned to @ryang1428

**Pull request merged**